File: app/src/main/java/com/seafile/seadroid2/framework/model/GroupItemModel.java
package com.seafile.seadroid2.framework.model;

import androidx.annotation.StringRes;

import com.blankj.utilcode.util.CollectionUtils;
import com.seafile.seadroid2.SeadroidApplication;
import com.seafile.seadroid2.framework.db.entities.RepoModel;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class GroupItemModel extends BaseModel {
    @StringRes
    private int name;

    public String title;

    public boolean is_expanded = true;

    //
    public final List<RepoModel> repo_list = new ArrayList<>();

    public void addAllRepoList(List<RepoModel> repoList) {
        repo_list.clear();
        repo_list.addAll(repoList);
    }

    public List<RepoModel> getRepoList() {
        return repo_list;
    }

    public GroupItemModel(@StringRes int nameRes) {
        this.name = nameRes;
        checkable = false;
        title = SeadroidApplication.getAppString(nameRes);
    }

    public GroupItemModel(@StringRes int nameRes, List<RepoModel> repoList) {
        this.name = nameRes;
        if (CollectionUtils.isNotEmpty(repoList)) {
            repo_list.addAll(repoList);
        }
        checkable = false;
        title = SeadroidApplication.getAppString(nameRes);
    }

    public GroupItemModel(String title, List<RepoModel> repoList) {
        this.title = title;
        if (CollectionUtils.isNotEmpty(repoList)) {
            repo_list.addAll(repoList);
        }
        checkable = false;
    }
    public GroupItemModel(String title) {
        this.title = title;
        checkable = false;
    }

    public String getTitle() {
        return title;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        GroupItemModel that = (GroupItemModel) o;
        return Objects.equals(title, that.title);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(title);
    }
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/GroupItemModel.kt
package com.seafile.seadroid2.framework.model

import androidx.annotation.StringRes
import com.seafile.seadroid2.SeadroidApplication
import com.seafile.seadroid2.framework.db.entities.RepoModel

class GroupItemModel : BaseModel {
    @JvmField
    var title: String? = null

    @JvmField
    var is_expanded: Boolean = true

    @JvmField
    val repo_list: MutableList<RepoModel> = mutableListOf()

    constructor(@StringRes nameRes: Int) : super() {
        checkable = false
        title = SeadroidApplication.getAppString(nameRes)
    }

    constructor(@StringRes nameRes: Int, repoList: List<RepoModel>) : super() {
        repo_list.addAll(repoList)
        checkable = false
        title = SeadroidApplication.getAppString(nameRes)
    }

    constructor(title: String, repoList: List<RepoModel>) : super() {
        this.title = title
        repo_list.addAll(repoList)
        checkable = false
    }

    fun addAllRepoList(repoList: List<RepoModel>) {
        repo_list.clear()
        repo_list.addAll(repoList)
    }

    fun getRepoList(): List<RepoModel> = repo_list

    fun getTitle(): String? = title

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is GroupItemModel) return false
        return title == other.title
    }

    override fun hashCode(): Int = title?.hashCode() ?: 0
}
